### PR TITLE
IGNITE-22806 remove tables counters functionality

### PR DESCRIPTION
**Pull request by alievmirza**
https://issues.apache.org/jira/browse/IGNITE-22806

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java [hunk 1]
```diff
@@ -324,15 +324,6 @@ public static CompletableFuture<?>[] triggerAllTablePartitionsRebalance(
     /** Key prefix for switch append assignments. */
     public static final String ASSIGNMENTS_SWITCH_APPEND_PREFIX = "assignments.switch.append.";
```
Comment (sanpwc) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceUtil.java:325]: Curious about why we do use APPEND_PREFIX only in RebalanceUtil itself, switchAppendKey with package-private scope but do register watch fo REDUCE_PREFIX in TableManager. That's suspicious. Not related to your PR though.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 2]
```diff
@@ -1225,6 +1213,28 @@ private PartitionReplicaListener createReplicaListener(
         );
     }
 
+    private CompletableFuture<Set<Assignment>> calculateAssignments(TablePartitionId tablePartitionId) {
+        // TODO: IGNITE-22661 Potentially unsafe to use the latest catalog version, as the tables might not already present
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1217]: Nice, that you've moved todo here. Initially I though that you've just removed it entirely.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 3]
```diff
@@ -1180,8 +1168,8 @@ private RaftGroupEventsListener createRaftGroupEventsListener(int zoneId, TableP
                 replicaGrpId,
                 busyLock,
                 partitionMover,
-                rebalanceScheduler,
-                zoneId
```
Comment (sanpwc) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1184]: Method parameter zoneId is no longer used.

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceRaftGroupEventsListener.java [hunk 4]
```diff
@@ -407,31 +310,22 @@ static void doStableKeySwitch(
...
                     )
             ).get();
 
-            // TODO: IGNITE-22661 Potentially unsafe to use the latest catalog version, as the tables might not already present
-            //  in the catalog. Better to take the version from Assignments.
-            int catalogVersion = catalogService.latestCatalogVersion();
-
-            Set<Assignment> calculatedAssignments =
-                    calculateAssignments(tablePartitionId, catalogService, distributionZoneManager, catalogVersion).get();
+            Set<Assignment> calculatedAssignments = calculateAssignmentsFn.apply(tablePartitionId).get();
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceRaftGroupEventsListener.java:317]: fixed

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceTest.java [hunk 5]
```diff
@@ -158,98 +148,6 @@ void assignmentsChangingOnNodeLeaveNodeJoin() throws Exception {
...
-        Set<String> allNodes = cluster.runningNodes().map(IgniteImpl::name).collect(Collectors.toSet());
-
-        for (Integer tableId : tableIds) {
-            waitForStableAssignmentsInMetastore(allNodes, tableId);
-        }
-
-        // Block low watermark change with an open ro tx.
-        cluster.aliveNode().transactions().begin(new TransactionOptions().readOnly(true));
-
-        alterZone(zoneName, 2);
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceTest.java:236]: removed

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceTest.java [hunk 6]
```diff
@@ -158,98 +148,6 @@ void assignmentsChangingOnNodeLeaveNodeJoin() throws Exception {
...
-
-        dropTestTable("TEST1");
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/table/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceTest.java:238]: removed

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java (hunk 3 above)
Comment (alievmirza) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:1184]: fixed

**Review by sanpwc — APPROVED** [review]

**Pull request merged**